# Overfitting single intervals

## imports

In [1]:
import os
import pandas as pd

## data

In [2]:
# paths

ROOT = ".."
RAW_DIR = os.path.join(ROOT,"data","raw")

track_metadata_file = os.path.join(RAW_DIR,"articles","Avsec2026","alphagenome_pytorch","track_metadata.parquet")

In [3]:
# loads

track_metadata = pd.read_parquet(track_metadata_file)

## select track_indices

In [4]:
track_metadata

,track_name,track_strand,ontology_curie,assay_title,biosample_name,biosample_type,organism,output_type,gtex_tissue,transcription_factor,histone_mark
0,CL:0000084 ATAC-seq,.,CL:0000084,ATAC-seq,T-cell,primary_cell,human,atac,NaN,NaN,NaN
1,CL:0000100 ATAC-seq,.,CL:0000100,ATAC-seq,motor neuron,in_vitro_differentiated_cells,human,atac,NaN,NaN,NaN
2,CL:0000236 ATAC-seq,.,CL:0000236,ATAC-seq,B cell,primary_cell,human,atac,NaN,NaN,NaN
3,CL:0000623 ATAC-seq,.,CL:0000623,ATAC-seq,natural killer cell,primary_cell,human,atac,NaN,NaN,NaN
4,CL:0000624 ATAC-seq,.,CL:0000624,ATAC-seq,"CD4-positive, alpha-beta T cell",primary_cell,human,atac,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...
12247,Padding,.,NaN,NaN,NaN,NaN,mouse,splice_sites_junction,NaN,NaN,NaN
12248,Padding,.,NaN,NaN,NaN,NaN,mouse,splice_sites_junction,NaN,NaN,NaN
12249,Padding,.,NaN,NaN,NaN,NaN,mouse,splice_sites_junction,NaN,NaN,NaN
12250,Padding,.,NaN,NaN,NaN,NaN,mouse,splice_sites_junction,NaN,NaN,NaN


In [5]:
track_metadata.query("track_name!='Padding'").value_counts(["organism","output_type"])

organism  output_type                
human     chip_tf                        1617
          chip_histone                   1116
          splice_sites_usage              734
          rna_seq                         667
          cage                            546
          splice_sites_junction           367
          dnase                           305
mouse     cage                            188
          chip_histone                    183
          splice_sites_usage              180
          rna_seq                         173
human     atac                            167
mouse     chip_tf                         127
          splice_sites_junction            90
          dnase                            67
human     pair_activations                 28
mouse     atac                             18
human     procap                           12
mouse     pair_activations                  8
human     splice_sites_classification       4
mouse     splice_sites_classification     

In [6]:
track_metadata.query("output_type=='rna_seq' & organism=='human' & track_name!='Padding'")

,track_name,track_strand,ontology_curie,assay_title,biosample_name,biosample_type,organism,output_type,gtex_tissue,transcription_factor,histone_mark
1408,CL:0000047 polyA plus RNA-seq,+,CL:0000047,polyA plus RNA-seq,neuronal stem cell,in_vitro_differentiated_cells,human,rna_seq,,NaN,NaN
1409,CL:0000062 total RNA-seq,+,CL:0000062,total RNA-seq,osteoblast,primary_cell,human,rna_seq,,NaN,NaN
1410,CL:0000084 polyA plus RNA-seq,+,CL:0000084,polyA plus RNA-seq,T-cell,primary_cell,human,rna_seq,,NaN,NaN
1411,CL:0000084 total RNA-seq,+,CL:0000084,total RNA-seq,T-cell,primary_cell,human,rna_seq,,NaN,NaN
1412,CL:0000115 total RNA-seq,+,CL:0000115,total RNA-seq,endothelial cell,in_vitro_differentiated_cells,human,rna_seq,,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...
2070,UBERON:0018115 polyA plus RNA-seq,.,UBERON:0018115,polyA plus RNA-seq,left renal pelvis,tissue,human,rna_seq,,NaN,NaN
2071,UBERON:0018116 polyA plus RNA-seq,.,UBERON:0018116,polyA plus RNA-seq,right renal pelvis,tissue,human,rna_seq,,NaN,NaN
2072,UBERON:0018117 polyA plus RNA-seq,.,UBERON:0018117,polyA plus RNA-seq,left renal cortex interstitium,tissue,human,rna_seq,,NaN,NaN
2073,UBERON:0018118 polyA plus RNA-seq,.,UBERON:0018118,polyA plus RNA-seq,right renal cortex interstitium,tissue,human,rna_seq,,NaN,NaN


In [12]:
track_metadata.query("output_type=='rna_seq' & organism=='human' & track_name!='Padding' & track_strand=='.'")

,track_name,track_strand,ontology_curie,assay_title,biosample_name,biosample_type,organism,output_type,gtex_tissue,transcription_factor,histone_mark
1950,CL:0000100 total RNA-seq,.,CL:0000100,total RNA-seq,motor neuron,in_vitro_differentiated_cells,human,rna_seq,,NaN,NaN
1951,CL:0000121 polyA plus RNA-seq,.,CL:0000121,polyA plus RNA-seq,Purkinje cell,primary_cell,human,rna_seq,,NaN,NaN
1952,CL:0000182 polyA plus RNA-seq,.,CL:0000182,polyA plus RNA-seq,hepatocyte,in_vitro_differentiated_cells,human,rna_seq,,NaN,NaN
1953,CL:0000187 polyA plus RNA-seq,.,CL:0000187,polyA plus RNA-seq,myocyte,in_vitro_differentiated_cells,human,rna_seq,,NaN,NaN
1954,CL:0000192 polyA plus RNA-seq,.,CL:0000192,polyA plus RNA-seq,smooth muscle cell,in_vitro_differentiated_cells,human,rna_seq,,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...
2070,UBERON:0018115 polyA plus RNA-seq,.,UBERON:0018115,polyA plus RNA-seq,left renal pelvis,tissue,human,rna_seq,,NaN,NaN
2071,UBERON:0018116 polyA plus RNA-seq,.,UBERON:0018116,polyA plus RNA-seq,right renal pelvis,tissue,human,rna_seq,,NaN,NaN
2072,UBERON:0018117 polyA plus RNA-seq,.,UBERON:0018117,polyA plus RNA-seq,left renal cortex interstitium,tissue,human,rna_seq,,NaN,NaN
2073,UBERON:0018118 polyA plus RNA-seq,.,UBERON:0018118,polyA plus RNA-seq,right renal cortex interstitium,tissue,human,rna_seq,,NaN,NaN


In [7]:
(
    track_metadata
    .query("output_type=='rna_seq' & organism=='human' & track_name!='Padding'")
    .reset_index(drop=True)
    .query("biosample_name.str.contains('K562')")
)

,track_name,track_strand,ontology_curie,assay_title,biosample_name,biosample_type,organism,output_type,gtex_tissue,transcription_factor,histone_mark
119,EFO:0002067 polyA plus RNA-seq,+,EFO:0002067,polyA plus RNA-seq,K562,cell_line,human,rna_seq,,NaN,NaN
120,EFO:0002067 total RNA-seq,+,EFO:0002067,total RNA-seq,K562,cell_line,human,rna_seq,,NaN,NaN
390,EFO:0002067 polyA plus RNA-seq,-,EFO:0002067,polyA plus RNA-seq,K562,cell_line,human,rna_seq,,NaN,NaN
391,EFO:0002067 total RNA-seq,-,EFO:0002067,total RNA-seq,K562,cell_line,human,rna_seq,,NaN,NaN
571,EFO:0002067 polyA plus RNA-seq,.,EFO:0002067,polyA plus RNA-seq,K562,cell_line,human,rna_seq,,NaN,NaN


In [8]:
(
    track_metadata
    .query("output_type=='splice_sites_usage' & organism=='human' & track_name!='Padding'")
    .reset_index(drop=True)
    .query("biosample_name.str.contains('K562')")
)

,track_name,track_strand,ontology_curie,assay_title,biosample_name,biosample_type,organism,output_type,gtex_tissue,transcription_factor,histone_mark
139,usage_EFO:0002067 polyA plus RNA-seq,+,EFO:0002067,polyA plus RNA-seq,K562,cell_line,human,splice_sites_usage,,NaN,NaN
140,usage_EFO:0002067 total RNA-seq,+,EFO:0002067,total RNA-seq,K562,cell_line,human,splice_sites_usage,,NaN,NaN
506,usage_EFO:0002067 polyA plus RNA-seq,-,EFO:0002067,polyA plus RNA-seq,K562,cell_line,human,splice_sites_usage,,NaN,NaN
507,usage_EFO:0002067 total RNA-seq,-,EFO:0002067,total RNA-seq,K562,cell_line,human,splice_sites_usage,,NaN,NaN


In [9]:
(
    track_metadata
    .query("output_type=='splice_sites_junction' & organism=='human' & track_name!='Padding'")
    .reset_index(drop=True)
    .query("biosample_name.str.contains('K562')")
)

,track_name,track_strand,ontology_curie,assay_title,biosample_name,biosample_type,organism,output_type,gtex_tissue,transcription_factor,histone_mark
139,junction_EFO:0002067 polyA plus RNA-seq,.,EFO:0002067,polyA plus RNA-seq,K562,cell_line,human,splice_sites_junction,,NaN,NaN
140,junction_EFO:0002067 total RNA-seq,.,EFO:0002067,total RNA-seq,K562,cell_line,human,splice_sites_junction,,NaN,NaN


In [24]:
(
    track_metadata
    .query("output_type=='splice_sites_junction' & organism=='human' & track_name!='Padding'")
    .reset_index(drop=True)
    .query("biosample_name.isin(['BLaER1','GM12878']) & assay_title=='polyA plus RNA-seq'")
)

,track_name,track_strand,ontology_curie,assay_title,biosample_name,biosample_type,organism,output_type,gtex_tissue,transcription_factor,histone_mark
153,junction_EFO:0002784 polyA plus RNA-seq,.,EFO:0002784,polyA plus RNA-seq,GM12878,cell_line,human,splice_sites_junction,,NaN,NaN
191,junction_EFO:0030018 polyA plus RNA-seq,.,EFO:0030018,polyA plus RNA-seq,BLaER1,cell_line,human,splice_sites_junction,,NaN,NaN


Interesting tracks for pretrained weights are polyA plus RNA-seq stranded:
- rna_seq: 119 (+) and 390 (-)
- splice_sites_usage: 139 (+) and 506 (-)
- splice_sites_junction: 139